tools/runners/run-unittests.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+
+# Copyright JS Foundation and other contributors, http://js.foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+import argparse
+import glob
+import os
+import subprocess
+import sys
+
+TERM_NORMAL = '\033[0m'
+TERM_RED = '\033[1;31m'
+TERM_GREEN = '\033[1;32m'
+
+def get_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-q', '--quiet', action='store_true',
+                        help='Only print out failing tests')
+    parser.add_argument('path',
+                        help='Path of test binaries')
+
+    script_args = parser.parse_args()
+    return script_args
+
+
+def get_unittests(path):
+    unittests = []
+    files = glob.glob(os.path.join(path, 'unit-*'))
+    for testfile in files:
+        if os.path.isfile(testfile) and os.access(testfile, os.X_OK):
+            if sys.platform != 'win32' or testfile.endswith(".exe"):
+                unittests.append(testfile)
+    unittests.sort()
+    return unittests
+
+
+def main(args):
+    unittests = get_unittests(args.path)
+    total = len(unittests)
+    if total == 0:
+        print("%s: no unit-* test to execute", args.path)
+        return 1
+
+    tested = 0
+    passed = 0
+    failed = 0
+
+    runtime = os.environ.get('RUNTIME')
+    test_cmd = [runtime] if runtime else []
+
+    for test in unittests:
+        tested += 1
+        testpath = os.path.relpath(test)
+        proc = subprocess.Popen(test_cmd + [test],
```

Comment:
There is a convenience method: `subprocess.check_output` it will return the output and throws an exception in case the execution failed (the return code is not zero). Maybe we could use that here?

Reply:
Already replaced with subprocess.check_output, it really looks better.